telethon/helpers.py

```diff
@@ -101,11 +101,11 @@ def retry_range(retries):
     not a zero or a positive integer value, the sequence will be
     infinite, otherwise it will end at `retries + 1`.
     """
-    yield 1
```

Comment:
The problem with this is that it's *retries*. For 0 retries, it should try once (1st try) and then retry no more (@cher-nov correct me if I'm wrong, but I think this was our rationale?). Now with 0 retries we won't even try to, say, connect, because the loop won't iterate even once.

Reply:
Yes, that's right that was the reason to yield 1 before while loop.
But this is breaking for the reconnects as you mentioned before.

Reply:
> @cher-nov correct me if I'm wrong, but I think this was our rationale?
Literally this, yes.